app/services/foreman_ansible/ansible_report_importer.rb

```diff
@@ -15,6 +15,23 @@ def host
           @host = Nic::Interface.find_by(:ip => hostname).host
         end
         super
+        partial_hostname_match(hostname)
+      end
+
+      def partial_hostname_match(hostname)
+        return @host unless @host.new_record?
+        hosts = Host.search_for("name ~ #{hostname}")
+        if hosts.count > 1
+          msg = "More than 1 host found for name #{hostname}, "
+          msg += 'please use host FQDN when uploading reports'
+          Rails.logger.warn msg
+          return @host
+        end
+        found_host hosts.first
```

Comment:
the code in method above also sets `@host = $found_host`, if we find a host based on search_for, should we set it there too?

also I wouldn't extract this logic to a separate method, just return `hosts.first || @host` or even better, `return @host if hosts.empty?` above the if